lib/location/method_cellular.c

```diff
@@ -0,0 +1,200 @@
+static struct lte_lc_cells_info cell_data = {
+	.neighbor_cells = neighbor_cells,
+};
+static bool lte_connected;
+static bool running;
+
+void method_cellular_lte_ind_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			lte_connected = false;
+			break;
+		}
+
+		LOG_DBG("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		lte_connected = true;
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS: {
+		struct lte_lc_cells_info cells = evt->cells_info;
+		struct lte_lc_cell cur_cell = cells.current_cell;
+
+		LOG_DBG("Cell measurements results received");
+		memset(&cell_data, 0, sizeof(struct lte_lc_cells_info));
+
+		if (cur_cell.id) {
+			/* Copy current and neighbor cell information */
+			memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
+			memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
+				sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+			cell_data.neighbor_cells = neighbor_cells;
+		} else {
+			LOG_INF("No current cell information from modem.");
+		}
+		if (!cells.ncells_count) {
+			LOG_INF("No neighbor cell information from modem.");
+		}
+		k_sem_give(&cellmeas_data_ready);
+	} break;
+	default:
+		break;
+	}
+}
+
+static int method_cellular_ncellmeas_start(void)
+{
+	int err;
+
+	LOG_DBG("Triggering start of cell measurements");
+	err = lte_lc_neighbor_cell_measurement();
+	if (err) {
+		LOG_ERR("Failed to initiate neighbor cell measurements: %d", err);
+		loc_core_event_cb_error();
+		running = false;
+		return err;
+	}
+	return 0;
+}
+
+static void method_cellular_positioning_work_fn(struct k_work *work)
+{
+	struct multicell_location location;
+	struct loc_location location_result = { 0 };
+	int ret;
+	struct k_work_args *work_data = CONTAINER_OF(work, struct k_work_args, work_item);
+	const struct loc_cellular_config cellular_config = work_data->cellular_config;
+
+	loc_core_timer_start(cellular_config.timeout);
+
+	ret = method_cellular_ncellmeas_start();
+	if (ret) {
+		LOG_WRN("Cannot start cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+		return;
+	}
+
+	ret = k_sem_take(&cellmeas_data_ready, K_SECONDS(60)); /* TODO: timeout to params? */
+
+	if (ret) {
+		LOG_WRN("Timeout for waiting cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+	} else if (running) {
+		if (cell_data.current_cell.id == 0) {
+			LOG_WRN("No cells were found - cannot fetch location");
+			loc_core_event_cb_error();
+			running = false;
+			return;
+		}
+
+		/* TODO: pass a device id / IMEI (fetched runtime?) */
+		ret = multicell_location_get(&cell_data, NULL, &location);
```

Comment:
Timeout currently not working for HTTP requests so both cellular and WLAN REST requests

Reply:
@jhirsi What we discussed yesterday about using the timeout from loc_cellular_config won't work because REST timeouts are compile time configs. I don't know if we should just remove timeouts from cellular and wifi configs as they don't do much. They can basically stop the operation after scanning the cells but not during the REST call.